ssl/statem/statem_dtls.c

```diff
@@ -235,8 +235,7 @@ int dtls1_do_write(SSL *s, int type)
 
     if (s->write_hash) {
         if (s->enc_write_ctx
-            && ((EVP_CIPHER_CTX_mode(s->enc_write_ctx) == EVP_CIPH_GCM_MODE) ||
-                (EVP_CIPHER_CTX_mode(s->enc_write_ctx) == EVP_CIPH_CCM_MODE)))
+            && (EVP_CIPHER_CTX_flags(s->enc_write_ctx) & EVP_CIPH_FLAG_AEAD_CIPHER) != 0)
```

Comment:
Just split it on the `&` and indent to be in line with the "E" of `EVP_CIPHER_CTX_flags`. We do that sort of stuff all over the place.